### DISCUSSION TITLE
How to find the rotation matrix in a corotational formulation?

### DISCUSSION THREAD

**Discussion by ZhengZQ007** [OpenRadioss/OpenRadioss#873] [Contributors]
In a corotational formulation, we need first to compute the rotation matrix [R]. In continuum mechanics, R is obtained through deformation gradient F. But I find the algorithm used by  OpenRadioss to compute R seems different. In the HEPH brick element, the transformation matrix seems to be calculated by subroutines SREPISO3 and SORTHO3. Could you explain the theory behind the algorithm?

**Comment by qiangRadioss** [comment]
Yes, Radioss does a different way than using [R] by decomposing [F] for brick element.
First reason is the cost of computation, second is the memory that the initial configuration has not to be saved.
We can say that the two methods are equivalent if the element follows well the rotation (local element system is well defined) that the objectivity w.r.t to rigid rotation has been respected.
The key is the local element system (e1,e2,e3), the algorithm consists to find the position where the angle between ej (j=1,3) and (r,s,t) are equal.
hope this helps.

**Reply by ZhengZQ007** [reply]
Thanks for the answer, it's very helpful for me.
